xbmc/filesystem/ShoutcastFile.cpp

```diff
@@ -66,12 +66,18 @@ bool CShoutcastFile::Open(const CURL& url)
   bool result = m_file.Open(url2);
   if (result)
   {
-    m_tag.SetTitle(m_file.GetHttpHeader().GetValue("icy-name"));
-    if (m_tag.GetTitle().empty())
-      m_tag.SetTitle(m_file.GetHttpHeader().GetValue("ice-name")); // icecast
+    std::string icyTitle;
+    icyTitle = m_file.GetHttpHeader().GetValue("icy-name");
+    if (icyTitle.empty())
+      icyTitle = m_file.GetHttpHeader().GetValue("ice-name"); // icecast
     m_tag.SetGenre(m_file.GetHttpHeader().GetValue("icy-genre"));
     if (m_tag.GetGenre().empty())
       m_tag.SetGenre(m_file.GetHttpHeader().GetValue("ice-genre")); // icecast
+    // Handle no title or badly set up servers where the default icy-name header hasn't been changed
+    if (icyTitle.empty() || (icyTitle == "This is my server name"))
+      icyTitle = "shoutcast"; // Dummy value used by CMusicGUIInfo.cpp to indicate no station name
```

Comment:
Then it would display "shoutcast", which would be correct.  To be clear though, the code in guiinfo would get that from the items sortlabel but it would be the same value.  I don't see the point of creating another musicinfotag to use as a boolean flag indicating whether or not there is already a title set when the string can be used for the same purpose. Maybe returning an empty string would be enough.  I can test that and see.

Reply:
Ah no.  Just realised that if I do that there could be instances when Kodi is playing a stream that isn't shoutcast.  In that case, that label _will_ be empty and it would get set to some unknown value (from the sortlabel) if the check were just for an empty string.